### DISCUSSION TITLE
LCP, CLS not available with Playwright headless mode

### DISCUSSION THREAD

**Discussion by ericlovetocode** [artilleryio/artillery#1902] [Artillery]
I'm doing load testing with the Playwright engine. I found on both Windows Server and Ubuntu the LCP and CLS metrics are not available in headless mode.

Here's the version of Artillery that I'm using:
- artillery/2.0.0-31 win32-x64 node-v18.14.0
- artillery/2.0.0-31 linux-x64 node-v18.16.0

I'm new to web browser load testing. Appreciate any help you can give. Thanks!

**Comment by hassy** [comment]
thanks @ericlovetocode! does Artillery report any/all other metrics as expected? (so only LCP and CLS are missing)

**Comment by ericlovetocode** [comment]
Thanks @hassy for the reply!

Yes, other metrics are shown as expected. Only LCP and CLS are missing. Please see screenshots below:

Running headed:

![image](https://user-images.githubusercontent.com/32584915/232583266-30bd63d7-810c-4891-9e08-fa0f0168ac2e.png)

Running headless:

![image](https://user-images.githubusercontent.com/32584915/232583323-d8680e08-3507-45a9-b369-ab65d561cf0b.png)

**Comment by ericlovetocode** [comment]
I may have figured it out. Here's what I did to get LCP and CLS on Ubuntu:

1. Install xvfb (apt install xvfb)
2. Set headless to false in the YAML file
3. Prefix artillery run with xvfb-run (xvfb-run artillery run myYamlFile.yml)

I don't see any Chrome browser in my virtual desktop. Nevertheless, Playwright did not complain. The LCP metric seems to match my custom metric. Please let me know if I did anything wrong. Thanks!

**Comment by hassy** [comment]
@ericlovetocode thanks for sharing the solution! Curious - did you try running with the [Dockerfile](https://github.com/artilleryio/artillery-engine-playwright/blob/main/Dockerfile) we provide?
